include/clangmetatool/propagation/constant_cstring_propagator.h

```diff
@@ -0,0 +1,93 @@
+#ifndef INCLUDED_CLANGMETATOOL_PROPAGATION_CONSTANT_CSTRING_PROPAGATOR_H
+#define INCLUDED_CLANGMETATOOL_PROPAGATION_CONSTANT_CSTRING_PROPAGATOR_H
+
+#include <iostream>
+#include <string>
+
+/**
+ * Forward declarations for clang types
+ */
+namespace clang {
+  class CompilerInstance;
+  class DeclRefExpr;
+  class FunctionDecl;
+}
+
+namespace clangmetatool {
+namespace propagation {
+
+/**
+ * Forward declaration to implementation details of the propagator.
+ */
+class ConstantCStringPropagatorImpl;
+
+/**
+ * ConstantCStringPropagator is a tool to run a propagation
+ * over the string variables that are used within functions
+ * and attempt to determine the string values of those
+ * variables.
+ *
+ * The analysis runs once per function even if runPropagation
+ * is called multiple times.
+ *
+ * It is also important to note that this analysis assumes
+ * that all loops will only be be run through the first time
+ * and effectively ignores any changes that may be made from
+ * reentering a block from the loop.
+ */
+class ConstantCStringPropagator {
+private:
+  /**
+    * Pointer to implementation.
+    */
+  ConstantCStringPropagatorImpl* impl;
+
+public:
+  /**
+   * Explicit constructor to allow for implementation details.
+   *    - ci is a pointer to an instance of the clang compiler
+   */
+  ConstantCStringPropagator(const clang::CompilerInstance* ci);
+
+  /**
+    * Explicit destructor.
+    */
+  ~ConstantCStringPropagator();
+
+  /**
+   * Given the surrounding function and the usage of a char*
+   * holding variable, attempt to determine the value held by that variable.
+   *
+   * This will return '<UNRESOLVED>' if a deterministic value cannot be
+   * determined for the variable.
+   */
+  std::string runPropagation
```

Comment:
hmm... I wonder if this could return a `const char*` instead such that it could return NULL if it's unresolved.

Reply:
This would be easy, will do.